src/base/default_allocator.rs

```diff
@@ -96,6 +130,32 @@ impl<T: Scalar, C: Dim> Allocator<T, Dynamic, C> for DefaultAllocator {
 
         VecStorage::new(nrows, ncols, res)
     }
+
+    #[inline]
+    fn allocate_from_row_iterator<I: IntoIterator<Item = T>>(
+        nrows: Dynamic,
+        ncols: C,
+        iter: I,
+    ) -> Self::Buffer {
+        let it = iter.into_iter().take(nrows.value() * ncols.value());
+        let mut res: Vec<T> = Vec::with_capacity(nrows.value() * ncols.value());
+        let res_ptr = res.as_mut_ptr();
+        let mut count = 0;
+
+        unsafe {
+            for (i, e) in it.enumerate() {
+                *res_ptr.add((i % ncols.value()) * nrows.value() + i / ncols.value()) = e;
```

Comment:
This is unsound because it will run the destructor of the uninitialized entry we are overwriting.
I think this constructior can be constructed using something like this:
1. Allocate the uninitialized storage using `Self::new_uninitialized`.
2. Initialize the storage’s entries using `MaybeUninit`.
3. Call `Self::assume_init` on the storage.